verilog/analysis/checkers/always_ff_only_local_blocking_rule.cc

```diff
@@ -0,0 +1,167 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/always_ff_only_local_blocking_rule.h"
+
+#include <set>
+#include <string>
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "common/analysis/citation.h"
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/bound_symbol_manager.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/analysis/syntax_tree_search.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::GetStyleGuideCitation;
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::SearchSyntaxTree;
+using verible::SyntaxTreeContext;
+using verible::matcher::Matcher;
+
+// Register AlwaysFFOnlyLocalBlockingRule
+VERILOG_REGISTER_LINT_RULE(AlwaysFFOnlyLocalBlockingRule);
+
+absl::string_view AlwaysFFOnlyLocalBlockingRule::Name() {
+  return "always-ff-only-local-blocking";
+}
+const char AlwaysFFOnlyLocalBlockingRule::kTopic[] = "sequential-logic";
+const char AlwaysFFOnlyLocalBlockingRule::kMessage[] =
+    "Use blocking assignments only for locals inside \'always_ff\' sequential blocks.";
+
+std::string AlwaysFFOnlyLocalBlockingRule::GetDescription(
+    DescriptionType description_type) {
+  return absl::StrCat("Checks that there are no occurrences of ",
+                      "blocking assignment to non-locals in sequential logic.");
+}
+
+
+static bool const  DEBUG = false;
+
+void AlwaysFFOnlyLocalBlockingRule::HandleSymbol(verible::Symbol const& symbol,
+                                           SyntaxTreeContext const& context) {
+
+  static Matcher const  always_ff_matcher = NodekAlwaysStatement(AlwaysFFKeyword());
```

Comment:
Instead of assignment, maybe call constructor directly (the rest of the codebase is using this style, so this keeps it consistent):

```c++
  static const Matcher always_ff_matcher(NodekAlwaysStatement(AlwaysFFKeyword());
```